Composer/packages/client/src/components/CreationFlow/index.tsx

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import Path from 'path';
+
+import React, { useEffect, useContext, useRef, Fragment, Suspense } from 'react';
+import { RouteComponentProps, Router, navigate } from '@reach/router';
+
+import { LoadingSpinner } from '../../components/LoadingSpinner';
+import { CreationFlowStatus } from '../../constants';
+import { StoreContext } from '../../store';
+import Home from '../../pages/home';
+
+import { CreateOptions } from './CreateOptions/index';
+import { OpenProject } from './OpenProject';
+import DefineConversation from './DefineConversation';
+
+type CreationFlowProps = RouteComponentProps<{}>;
+
+const CreationFlow: React.FC<CreationFlowProps> = () => {
+  const { state, actions } = useContext(StoreContext);
+  const { creationFlowStatus } = state;
+  const {
+    fetchTemplates,
+    openBotProject,
+    createProject,
+    saveProjectAs,
+    saveTemplateId,
+    fetchStorages,
+    fetchFolderItemsByPath,
+    setCreationFlowStatus,
+  } = actions;
+  const { templateId, templateProjects, storages } = state;
+  const currentStorageIndex = useRef(0);
+  const storage = storages[currentStorageIndex.current];
+  const currentStorageId = storage ? storage.id : 'default';
+  useEffect(() => {
+    if (storages && storages.length) {
+      const storageId = storage.id;
+      const path = storage.path;
+      const formattedPath = Path.normalize(path);
+      fetchFolderItemsByPath(storageId, formattedPath);
+    }
+  }, [storages]);
+
+  const init = async () => {
+    (async function() {
```

Comment:
This doesn't make sense to me. Why do we need to double wrap this init function?